apps/src/templates/manageStudents/manageStudentsRedux.js

```diff
@@ -240,43 +282,39 @@ export default function manageStudents(state=initialState, action) {
     };
   }
   if (action.type === ADD_STUDENT_FAILURE) {
-    return {
+    let newState = {
       ...state,
-      studentData: {
-        ...state.studentData,
-        [action.studentId]: {
-          ...state.studentData[action.studentId],
-          isSaving: false,
-        }
-      },
-      addStatus: AddStatus.FAIL
+      addStatus: {status: AddStatus.FAIL, numStudents: action.numStudents}
     };
+    for (let i = 0; i<action.studentIds.length; i++) {
+      newState.studentData[action.studentIds[i]] = {
+        ...state.studentData[action.studentIds[i]],
+        isSaving: false,
+      };
+    }
+    return newState;
   }
   if (action.type === ADD_STUDENT_SUCCESS) {
-    // omit action.rowId
-    return {
+    let newState = {
       ...state,
       studentData: {
-        [action.studentData.id]: {
-          ...action.studentData,
-          loginType: state.loginType,
-          sectionId: state.sectionId,
-        },
-        ..._.omit(state.studentData, action.rowId),
+        ..._.omit(state.studentData, action.rowIds),
+        ...action.studentData,
         [addRowId]: {
           ...blankAddRow,
-          loginType: state.loginType
-        },
+          loginType: state.loginType,
+        }
       },
       editingData: {
-        ..._.omit(state.editingData, action.rowId),
+        ..._.omit(state.editingData, action.rowIds),
         [addRowId]: {
           ...blankAddRow,
-          loginType: state.loginType
+          loginType: state.loginType,
         }
       },
-      addStatus: AddStatus.SUCCESS,
+      addStatus: {status: AddStatus.SUCCESS, numStudents: action.numStudents}
     };
```

Comment:
When should we have spaces in braces { like this } and when should we not {like this}? I see both in our codebase, and I don't know think the styleguide has a ruling. 🤷🏼‍♀️ just curious

Reply:
Apparently they are both okay: https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#js-object-curly-spacing